modules/remotebackend/remotebackend.cc

```diff
@@ -564,78 +508,55 @@ bool RemoteBackend::deleteTSIGKey(const DNSName& name) {
 
-   // expect array
-   if (answer["result"].IsArray()) {
-      for(rapidjson::Value::ValueIterator iter = answer["result"].Begin(); iter != answer["result"].End(); iter++) {
-         struct TSIGKey key;
-         rapidjson::Value value;
-         value = "";
-         key.name = DNSName(getString(JSON_GET((*iter), "name", value)));
-         key.algorithm = DNSName(getString(JSON_GET((*iter), "algorithm", value)));
-         key.key = getString(JSON_GET((*iter), "content", value));
-
-         if (key.name.empty() ||
-             key.algorithm.empty() ||
-             key.key.empty())
-           throw PDNSException("Invalid reply for getTSIGKeys query");
-
-         keys.push_back(key);
-      }
+   for(const auto& jsonKey: answer["result"].array_items()) {
+     struct TSIGKey key;
+     key.name = DNSName(stringFromJson(jsonKey, "name"));
+     key.algorithm = DNSName(stringFromJson(jsonKey, "algorithm"));
+     key.key = stringFromJson(jsonKey, "content");
+     keys.push_back(key);
    }
 
    return true;
 }
 
 bool RemoteBackend::getDomainInfo(const DNSName& domain, DomainInfo &di) {
-   rapidjson::Document query,answer;
-   rapidjson::Value parameters;
-   rapidjson::Value value;
-   std::string kind;
-
-   query.SetObject();
-   JSON_ADD_MEMBER(query, "method", "getDomainInfo", query.GetAllocator());
-   parameters.SetObject();
-   JSON_ADD_MEMBER_DNSNAME(parameters, "name", domain, query.GetAllocator());
-   query.AddMember("parameters", parameters, query.GetAllocator());
+   if (domain.empty()) return false;
+   Json query = Json::object{
+     { "method", "getDomainInfo" },
+     { "parameters", Json::object {
+       { "name", domain.toString() }
+     }}
+   };
 
+   Json answer;
    if (this->send(query) == false || this->recv(answer) == false)
      return false;
 
-   // make sure we got zone & kind
-   if (!answer["result"].IsObject() || !answer["result"].HasMember("zone")) {
-      L<<Logger::Error<<kBackendId<<" Missing zone in getDomainInfo return value"<<endl;
-      throw PDNSException();
-   }
-   value = -1;
-   // parse return value. we need at least zone,serial,kind
-   di.id = getInt(JSON_GET(answer["result"],"id",value));
-   di.zone = DNSName(getString(answer["result"]["zone"]));
-
-   if (answer["result"].HasMember("masters") && answer["result"]["masters"].IsArray()) {
-     rapidjson::Value& value = answer["result"]["masters"];
-     for(rapidjson::Value::ValueIterator i = value.Begin(); i != value.End(); i++) {
-        di.masters.push_back(getString(*i));
-     }
+   di.id = intFromJson(answer["result"], "id", -1);
+   di.zone = DNSName(stringFromJson(answer["result"], "zone"));
+   for(const auto& master: answer["result"]["masters"].array_items())
+     di.masters.push_back(master.string_value());
+
+   di.notified_serial = static_cast<unsigned int>(doubleFromJson(answer["result"], "notified_serial", -1));
+   di.serial = static_cast<unsigned int>(doubleFromJson(answer["result"], "serial", 0));
```

Comment:
my style would be to use the json11-provided functions when they do what i want, and only use wrappers to do other stuff. in this case, you exactly want the json11 default behaviour, so going to the wrapper is rather pointless. but not my code ;-)